coral/src/app/features/topics/details/overview/TopicOverview.tsx

```diff
@@ -29,6 +90,51 @@ function TopicOverview() {
           </Box.Flex>
         </Card>
       </GridItem>
+      {showRequestPromotionBanner && (
```

Comment:
I would move all related to the promotion banner into it's own component that takes the needed props. That will make testing a lot easier, since you don't have to mock endpoints for that but can test the different states based on their props. `SchemaPromotionModal` as example.

Reply:
Added it here [7b5ae77](https://github.com/aiven/klaw/pull/1420/commits/7b5ae774fe3ccee53864e842f03213b5813713f1)

Reply:
IOmproved it here haha [ab322e8](https://github.com/aiven/klaw/pull/1420/commits/ab322e8533d420e227eb5afb907a642a70a7c932)